canonical-web-and-design/maas-ui
# Populate repository add/edit fields with websocket data

## Done
- Populated RepositoryAdd and RepositoryEdit forms with packagerepository and general data

## QA
- Go to `:8000/MAAS/settings/repositories` and click "Add PPA".
- Check that the url field is already prefixed with `ppa:`, there is no section for "Distributions" or "Components", the title is "Add PPA" and the save button says "Save PPA"
- Click on "Add repository"
- Check that there is a section for "Distributions" and "Components", the title is "Add repository" and the save button says "Save repository"
- Add some repos at `:5240/MAAS/#/settings/repositories`
- Go to `:8000/MAAS/settings/repositories`, click the edit icon and check that the repo data is correct
- Click the edit button for the first repo (default one)
- Check that there are checkboxes for "Disabled pockets" and "Disabled components"

## Screenshots
### Add PPA
![10 153 61 90_8000_MAAS_settings_repositories_add_ppa](https://user-images.githubusercontent.com/25733845/63705519-9deab900-c82d-11e9-96ac-9faaffd5b5fb.png)

### Add repository
![10 153 61 90_8000_MAAS_settings_repositories_add_ppa (1)](https://user-images.githubusercontent.com/25733845/63705544-ad6a0200-c82d-11e9-9cda-6a1e098b39ae.png)

### Edit PPA
![10 153 61 90_8000_MAAS_settings_repositories (2)](https://user-images.githubusercontent.com/25733845/63705459-7b58a000-c82d-11e9-827c-a68b1a9be85f.png)

### Edit repository
![10 153 61 90_8000_MAAS_settings_repositories_add_ppa (2)](https://user-images.githubusercontent.com/25733845/63705557-b4911000-c82d-11e9-8bfa-96b4cdb9b09c.png)

### Edit default repository
![10 153 61 90_8000_MAAS_settings_repositories (3)](https://user-images.githubusercontent.com/25733845/63705568-ba86f100-c82d-11e9-8a08-e0eac713353a.png)

### Mobile view
![10 153 61 90_8000_MAAS_settings_repositories (4)](https://user-images.githubusercontent.com/25733845/63705570-bf4ba500-c82d-11e9-9948-79dde5106b51.png)


Fixes canonical-web-and-design/maas-squad#1431
Fixes canonical-web-and-design/maas-squad#1433
